src/hotspot/share/gc/shenandoah/shenandoahHeap.cpp

```diff
@@ -1337,10 +1325,180 @@ void ShenandoahHeap::object_iterate(ObjectClosure* cl) {
   }
 
   assert(oop_stack.is_empty(), "should be empty");
+  // Reclaim bitmap
+  reclaim_aux_bitmap_for_iteration();
+}
+
+bool ShenandoahHeap::prepare_aux_bitmap_for_iteration() {
+    assert(SafepointSynchronize::is_at_safepoint(), "safe iteration is only available during safepoints");
+
+    if (!_aux_bitmap_region_special && !os::commit_memory((char*)_aux_bitmap_region.start(), _aux_bitmap_region.byte_size(), false)) {
+      log_warning(gc)("Could not commit native memory for auxiliary marking bitmap for heap iteration");
+      return false;
+    }
 
-  if (!_aux_bitmap_region_special && !os::uncommit_memory((char*)_aux_bitmap_region.start(), _aux_bitmap_region.byte_size())) {
-    log_warning(gc)("Could not uncommit native memory for auxiliary marking bitmap for heap iteration");
+    // Reset bitmap
+    _aux_bit_map.clear();
```

Comment:
Wait a sec, we have just uncommitted the aux bitmap, so `clear()` is bound to crash?

Reply:
No, here the aux_bit_map.clear() is called after commit_memory(), so clear() is safe to reset the bitmap.
the uncommit_memory() was moved to reclaim_aux_bitmap_for_iteration() at line 1354